back/engines/commercial/analytics/app/services/analytics/populate_dimensions_service.rb

```diff
@@ -10,6 +10,19 @@ def run
         populate_referrer_types
       end
 
+      def create_dates(from, to)
+        (from..to).each do |date|
+          Analytics::DimensionDate.create!(
```

Comment:
```suggestion
          Analytics::DimensionDate.find_or_create_by!(
```
https://apidock.com/rails/v6.1.3.1/ActiveRecord/Relation/find_or_create_by%21

Probably, it would not need to catch the error and would work a little bit faster, but in general, it's the same.

Reply:
It seems for most tenants it'll be just one create call per run, so performance is not important here.

~~I'm wondering how drastically it'll affect performance. So, for every tenant, we find or try to create a dimension date for almost every day since the platform got created, don't we?~~

~~It can mean thousand(-s) find/create requests per tenant. Let's say one find is 3 ms, and one create is 5 ms (very roughly). Then it can add up to 1-5 seconds per tenant. The Europe cluster has 661 tenants, which means a good few minutes per run.~~

~~Maybe a simple check like~~
```ruby
Analytics::PopulateDimensionsService.create_dates(from, to) if Analytics::DimensionDate.find_by(date: to).nil?
```
~~could solve it. But probably it's not a big deal.~~